pinot-core/src/main/java/com/linkedin/pinot/core/periodictask/PeriodicTaskScheduler.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (C) 2014-2018 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.periodictask;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.PriorityBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class PeriodicTaskScheduler {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PeriodicTaskScheduler.class);
+  private PriorityBlockingQueue<PeriodicTask> _periodicTasks;
+  private ScheduledExecutorService _executorService;
+  private long _intervalSeconds;
+  private long _initialDelaySeconds;
+
+  public PeriodicTaskScheduler() {
+    _executorService = Executors.newSingleThreadScheduledExecutor(runnable -> {
+      Thread thread = new Thread(runnable);
+      thread.setName("PeriodicTaskSchedulerExecutorService");
+      return thread;
+    });
+    _initialDelaySeconds = 120L;
+    _intervalSeconds = 300L;
+    _periodicTasks = new PriorityBlockingQueue<>(10, (o1, o2) -> {
+      if (o1.getExecutionTime() == o2.getExecutionTime()) {
+        return 0;
+      }
+      return o1.getExecutionTime() < o2.getExecutionTime() ? -1 : 1;
+    });
+  }
+
+  public void start() {
```

Comment:
Recommend changing this to **start(List\<PeriodicTask\> periodicTasks)** so that we don't need to add/remove each of the tasks